lib/awful/layout/init.lua

```diff
@@ -164,8 +164,14 @@ function layout.parameters(t, screen)
         gap_single_client = t.gap_single_client
     end
 
-    local min_clients       = gap_single_client and 1 or 2
-    local useless_gap       = t and (#clients >= min_clients and t.gap or 0) or 0
+    local useless_gap = 0
+    if (t) then
+        local l = layout.get(screen)
+        if (gap_single_client or not l.skip_gap and #clients >= 2
+                or l.skip_gap and not l.skip_gap(#clients, t)) then
+            useless_gap = t.gap
+        end
```

Comment:
Since we are proposing not-all-that-important code changes:
```lua
local useless_gap = 0
if t then
  local skip_gap = layout.get(screen).skip_gap or function(nclients)
    return nclients >= 2
  end 
  if gap_single_client or not skip_gap(#clients, t) then
    useless_gap = t.gap
  end
end
```
I'll leave it for others to decide which version is simpler.